s3/src/main/scala/geotrellis/spark/io/s3/S3RDDWriter.scala

```diff
@@ -1,32 +1,36 @@
 package geotrellis.spark.io.s3
 
-import geotrellis.spark.io.avro.codecs.KeyValueRecordCodec
-import org.apache.spark.rdd.RDD
 
-import java.io.{ObjectOutputStream, ByteArrayOutputStream, ByteArrayInputStream}
-import java.util.concurrent.Executors
-import com.amazonaws.services.s3.model.{AmazonS3Exception, PutObjectResult, ObjectMetadata, PutObjectRequest}
 import geotrellis.raster.Tile
 import geotrellis.spark._
 import geotrellis.spark.io._
 import geotrellis.spark.io.json._
-import geotrellis.spark.io.avro.{AvroRecordCodec, AvroEncoder}
+import geotrellis.spark.io.avro._
+import geotrellis.spark.io.avro.codecs.KeyValueRecordCodec
 import geotrellis.spark.io.index.{ZCurveKeyIndexMethod, KeyIndexMethod, KeyIndex}
 import geotrellis.spark.utils.KryoWrapper
+
+import com.amazonaws.services.s3.model.{AmazonS3Exception, PutObjectResult, ObjectMetadata, PutObjectRequest}
+import com.typesafe.scalalogging.slf4j._
+import org.apache.spark.rdd.RDD
+import scalaz.concurrent.Task
+import scalaz.stream.{Process, nondeterminism}
 import spray.json._
 import spray.json.DefaultJsonProtocol._
+
+import java.io.{ObjectOutputStream, ByteArrayOutputStream, ByteArrayInputStream}
+import java.util.concurrent.Executors
 import scala.reflect._
-import scalaz.concurrent.Task
-import scalaz.stream.{Process, nondeterminism}
-import com.typesafe.scalalogging.slf4j._
 
-class S3RDDWriter [K: AvroRecordCodec: ClassTag, V: AvroRecordCodec: ClassTag]() {
 
-  def getS3Client: () => S3Client = () => S3Client.default
-  val codec  = KeyValueRecordCodec[K, V]
-  val schema = codec.schema
+trait S3RDDWriter {
+
+  def getS3Client: () => S3Client
+
+  def write[K: AvroRecordCodec: ClassTag, V: AvroRecordCodec: ClassTag](rdd: RDD[(K, V)], bucket: String, keyPath: K => String, oneToOne: Boolean): Unit = {
```

Comment:
If we're going to remove `oneToOne` from accumulo RDD writer, should we do the same here?